Cleanup flyteplugins

Sovietaced opened the pull request:
## What changes were proposed in this pull request?

Cleans up unused constants, unused parameters, problematic handling of HTTP response bodies, bad string formats, etc.

## How was this patch tested?

Existing unit tests

## Check all the applicable boxes <!-- Follow the above conventions to check the box -->

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs

<!-- Add related pull requests for reviewers to check -->

## Docs link

<!-- Add documentation link built by CI jobs here, and specify the changed place --> 
 <div id='description'>
    <a href="https://bito.ai#summarystart"></a>
<h3>Summary by Bito</h3>
This pull request enhances the flyteplugins codebase by removing unused constants and parameters, fixing string formatting issues, and improving HTTP response body handling. These changes aim to boost code readability and maintainability while preserving existing functionality.
<!-- Disabling unit_tests and post_effort_to_review
<br>
<br>
<b>Unit tests added</b>: False
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 2
-->
</div>

Sovietaced commented on flyteplugins/go/tasks/plugins/webapi/connector/plugin.go:
```diff
@@ -295,11 +294,11 @@ func (p *Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phas
 
 	switch resource.Phase {
 	case flyteIdl.TaskExecution_QUEUED:
-		return core.PhaseInfoQueuedWithTaskInfo(time.Now(), core.DefaultPhaseVersion, resource.Message, taskInfo), nil
```
All the timestamps were unused

Sovietaced commented on flyteplugins/go/tasks/pluginmachinery/webapi/example/plugin.go:
```diff
@@ -8,16 +8,9 @@ import (
 	"github.com/flyteorg/flyte/flytestdlib/promutils"
 )
 
-const (
```
unused

Sovietaced commented on flyteplugins/go/tasks/plugins/hive/execution_state.go:
```diff
@@ -300,15 +300,15 @@ func mapLabelToPrimaryLabel(ctx context.Context, quboleCfg *config.Config, label
 	// which is determined specifically for the readability of the corresponding configmap yaml file
 	for _, clusterCfg := range quboleCfg.ClusterConfigs {
 		for _, l := range clusterCfg.Labels {
-			if label != "" && l == label {
```
Condition is always true because of previous assertion

Sovietaced commented on flyteplugins/go/tasks/plugins/array/k8s/management_test.go:
```diff
@@ -565,13 +565,6 @@ func TestCheckSubTasksState(t *testing.T) {
 func TestTerminateSubTasksOnAbort(t *testing.T) {
 	ctx := context.Background()
 	subtaskCount := 3
-	config := Config{
```
unused

Sovietaced commented on flyteplugins/go/tasks/pluginmachinery/flytek8s/container_helper_test.go:
```diff
@@ -377,8 +377,8 @@ func TestToK8sContainer(t *testing.T) {
-	inputReader.EXPECT().GetInputPath().Return(storage.DataReference("test-data-reference"))
-	inputReader.EXPECT().GetInputPrefixPath().Return(storage.DataReference("test-data-reference-prefix"))
+	inputReader.EXPECT().GetInputPath().Return("test-data-reference")
+	inputReader.EXPECT().GetInputPrefixPath().Return("test-data-reference-prefix")
```
A DataReference is a string so my edit was showing it as unnecessary

machichima commented on flyteplugins/go/tasks/config_load_test.go:
```diff
@@ -51,7 +51,7 @@ func TestLoadConfig(t *testing.T) {
 		}
 
-		assert.Equal(t, []v1.Toleration{tolGPU}, k8sConfig.ResourceTolerations[v1.ResourceName("nvidia.com/gpu")])
+		assert.Equal(t, []v1.Toleration{tolGPU}, k8sConfig.ResourceTolerations[("nvidia.com/gpu")])
```
Got it! Let's keep them removed

Sovietaced commented on flyteplugins/go/tasks/plugins/hive/execution_state.go:
```diff
@@ -412,9 +412,9 @@ func KickOffQuery(ctx context.Context, tCtx core.TaskExecutionContext, currentSt
 		if err != nil {
 			// This means that our cache has fundamentally broken... return a system error
-			logger.Errorf(ctx, "Cache failed to GetOrCreate for execution [%s] cache key [%s], owner [%s]. Error %s",
+			logger.Errorf(ctx, "Cache failed to GetOrCreate for execution [%v] cache key [%s], owner [%v]. Error %s",
```
Yes, just updated.

Sovietaced commented:
> Just went through some files and will continue on the remaining ones

Thanks for your thorough review. I believe I have addressed all of your feedback.

kumare3 commented:
Can we also drop some plugins like qubole, hive AWS batch (old) etc

Sovietaced commented:
> Can we also drop some plugins like qubole, hive AWS batch (old) etc

Say less, I love deleting old code :)

Sovietaced commented on flyteplugins/go/tasks/plugins/array/awsbatch/jobs_store.go:
```diff
@@ -254,7 +254,7 @@ func syncBatches(_ context.Context, client Client, handler EventHandler, batchCh
 		startIdx, endIdx := toRanges(len(jobIds), batchChunkSize)
 		for i := 0; i < len(startIdx); i++ {
-			logger.Debugf(ctx, "Syncing chunk [%v, %v) out of [%v] job ids.", startIdx[i], endIdx[i])
+			logger.Debugf(ctx, "Syncing chunk [%v, %v) out of [%v] job ids.", startIdx[i], endIdx[i], len(startIdx))
```
Done

machichima reviewed (approved):
LGTM!
